carcara/src/parser/mod.rs

```diff
@@ -1290,6 +1475,14 @@ impl<'a, R: BufRead> Parser<'a, R> {
 
                 Ok(result)
             }
+            Token::OpenParen => {
+                self.next_token()?;
+                self.expect_token(Token::ReservedWord(Reserved::Underscore))?;
+                let (op, op_args) = self.parse_indexed_operator()?;
+                let args = self.parse_sequence(Self::parse_term, true)?;
+                self.make_indexed_op(op, op_args, args)
+                    .map_err(|err| Error::Parser(err, head_pos))
```

Comment:
The tests are failing because, when adding support for parsing indexed operations (that start with the tokens `((`), we accidentally broke the parsing of `lambda` terms. This is a bit tricky to solve, but I think the cleanest way to do this without adding lookahead is this:
```suggestion
                if self.current_token == Token::ReservedWord(Reserved::Underscore) {
                    self.next_token()?;
                    let (op, op_args) = self.parse_indexed_operator()?;
                    let args = self.parse_sequence(Self::parse_term, true)?;
                    self.make_indexed_op(op, op_args, args)
                        .map_err(|err| Error::Parser(err, head_pos))
                } else {
                    let func = self.parse_application()?;
                    let args = self.parse_sequence(Self::parse_term, true)?;
                    self.make_app(func, args)
                        .map_err(|err| Error::Parser(err, head_pos))
                }
```

Reply:
ty